src/microbit/model/display.py

```diff
@@ -58,38 +75,47 @@ def show(self, value, delay=400, wait=True, loop=False, clear=False):
             )
             thread.start()
             return
-        while True:
-            if isinstance(value, Image):
-                self.__image = value.crop(
-                    0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
-                )
-            elif isinstance(value, (str, int, float)):
-                if isinstance(value, str):
-                    chars = list(value)
+
+        # Set current_pid to the thread's identifier
+        self.__current_pid = threading.get_ident()
+
+        images = []
+        use_delay = False
+        if isinstance(value, Image):
+            images.append(value.crop(0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT))
+        elif isinstance(value, (str, int, float)):
+            chars = list(str(value))
+            for c in chars:
+                images.append(Display.__get_image_from_char(c))
+            if len(chars) > 1:
+                use_delay = True
+        else:
+            # Check if iterable
+            try:
+                _ = iter(value)
+            except TypeError as e:
+                raise e
+
+            for elem in value:
+                if isinstance(elem, Image):
+                    images.append(
+                        elem.crop(0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT)
+                    )
+                elif isinstance(elem, str) and len(elem) == 1:
+                    images.append(Display.__get_image_from_char(elem))
+                # If elem is not char or image, break without iterating through rest of list
                 else:
-                    chars = list(str(value))
+                    break
+            use_delay = True
 
-                for c in chars:
-                    self.__image = Display.__get_image_from_char(c)
-                    time.sleep(delay / 1000)
-            else:
-                # Check if iterable
-                try:
-                    _ = iter(value)
-                except TypeError as e:
-                    raise e
-
-                for elem in value:
-                    if isinstance(elem, Image):
-                        self.__image = elem.crop(
-                            0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
-                        )
-                    elif isinstance(elem, str) and len(elem) == 1:
-                        self.__image = Display.__get_image_from_char(elem)
-                    # If elem is not char or image, break without iterating through rest of list
-                    else:
-                        break
+        while True:
+            for image in images:
+                self.__image = image
+                if use_delay:

```

Comment:
I might just remove this variable and always use the delay, not sure it's worth the extra lines of code.
If the original library doesn't block when it's only one image to show, the check `if len(images) >= 2` could be used instead.

Reply:
It blocks for 'delay' when it is an array of one image, but it doesnt block when it is just display.show(Image.HEART)